x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/NlpTokenizer.java

```diff
@@ -232,6 +234,19 @@ public TokenizationResult.Tokens tokenize(
                     tokenIdsSeq2 = tokenIdsSeq2.subList(0, maxSequenceLength() - extraTokens - tokenIdsSeq1.size());
                     tokenPositionMapSeq2 = tokenPositionMapSeq2.subList(0, maxSequenceLength() - extraTokens - tokenIdsSeq1.size());
                 }
+                case BALANCED -> {
```

Comment:
Because `NlpTokenizer` is an abstract class most of the unit testing of this code is in `BertTokenizerTests`. Look at `BertTokenizerTests::testTokenizeLargeInputMultiSequenceTruncation` and add a similar test to cover the `BALANCED` case

Reply:
thanks!